src/common/CMakeLists.txt

```diff
@@ -23,5 +23,8 @@ file(GLOB_RECURSE SOURCES
 
 set(OBJ_LIB ${LIB_NAME}_common)
 add_library(${OBJ_LIB} OBJECT ${SOURCES})
+set_target_properties(${OBJ_LIB} PROPERTIES COMPILE_FLAGS "${DNNL_ARCH_OPT_FLAGS}")
+target_link_libraries_build(${OBJ_LIB}
+    "${EXTRA_SHARED_LIBS};${EXTRA_STATIC_LIBS}")
```

Comment:
As I know `target_link_libraries` didn't work with object libraries before CMake 3.12 (DNNL supports 2.8.11+).

Can you please explain what is the goal of this change? Now (with 2.8.11) we can't propagate dependencies from object libraries and have to handle that globally in `src/CMakeLists.txt`. Do you see any issues with that?

Reply:
I didn't know that this is such a recent CMake feature.

We have a modified `TBB.cmake`, which sets the TBB include directories for the TBB targets, not globally, so we need to propagate these to the MKL-DNN libraries through `target_link_libraries_build`. With older MKL-DNN versions this worked fine because there was a single static/dynamic MKL-DNN library, and only one `target_link_libraries_build` was needed. Now this is more difficult due to the multiple object libraries. Is there any other way to propagate the TBB target properties to the MKL-DNN object libraries?

Reply:
We use global `include_directories` to propagate include directories from `TBB.cmake` to all source files:
https://github.com/intel/mkl-dnn/blob/master/cmake/TBB.cmake#L38

Doesn't it work for you? If you need to limit the scope you can call `include_directories` from a specific sub-directory.

Reply:
This doesn't work for us because it's not `include_directories` that we need to set but `INTERFACE_INCLUDE_DIRECTORIES`.

Reply:
Closing the discussion as the changes [are not relevant anymore](https://github.com/intel/mkl-dnn/pull/572#issuecomment-541941800)